backend/dataall/modules/dataset_sharing/services/share_object_service.py

```diff
@@ -590,3 +603,55 @@ def attach_dataset_folder_read_permission(session, share):
                 log.info(
                     f'Resource permission policy {DATASET_FOLDER_READ} to table {location.itemUri} for group {share.groupUri} already exists. Skip... '
                 )
+
+    @staticmethod
+    def get_share_logs_name_query(shareUri):
+        log.info(f'Get share Logs stream name for share {shareUri}')
+
+        query = """fields @logStream
+                        |filter  @message like 'bmm02skg'
+                        | sort @timestamp desc
+                        | limit 1
+                    """
+        return query
+
+    @staticmethod
+    def get_share_logs_query(log_stream_name):
+        query = f"""fields @timestamp, @message, @logStream, @log as @logGroup
+                    | sort @timestamp asc
+                    | filter @logStream like "{log_stream_name}"
+                    """
+        return query
+
+    @staticmethod
+    def get_share_logs(shareUri):
+        context = get_context()
+        if not ShareObjectService.check_view_log_permissions(context.username, context.groups, shareUri):
+            raise exceptions.ResourceUnauthorized(
+                username=context.username,
+                action='view logs',
```

Comment:
Would be a bit more verbose - `"View Share Logs"`

Reply:
Done)